checker/tests/resourceleak/MustCallAliasSubstitution.java

```diff
@@ -0,0 +1,16 @@
+// A test case that checks that resolving the obligations of one object and then
+// substituting a fresh object is not counted as an alias, for the purpose of MustCallAlias
+// verification.
+
+import java.io.*;
+import java.net.Socket;
+import org.checkerframework.checker.mustcall.qual.*;
+
+class MustCallAliasSubstitution {
```

Comment:
For illustrative purposes, it might be nice to extend this test case with a call to `example()`, where `close()` is invoked on the value returned from `example()`.  That will show how without reporting the error, the tool could allow a leak.

Reply:
b40c5ec38fb46385d34261762a2fc5ba602e3aa6